### PR TITLE
Block orientation

### PR DESCRIPTION
**Pull request by petterarvidsson**

### CODE REVIEW

**Comment by petterarvidsson** [comment]
This requires the changes to the server in https://github.com/konstructs/server/pull/134

File: lib/include/client.h [hunk 1]
```diff
@@ -67,7 +67,8 @@ namespace konstructs {
         void close_inventory();
         void talk(const string &text);
         void update_radius(const int radius);
-        void click_at(const int hit, const Vector3i pos, const int button, const int active);
+        void click_at(const int hit, const Vector3i pos, const int button, const int active,
+                      const uint8_t direction, const uint8_t rotation);
```
Comment (nsg) [inline lib/include/client.h]: rotation is never used in the implementation/in a call.

**Comment by nsg** [comment]
Nice, I haven't tried the code yet but I will later today. Looks really good, rotation opens up plenty of possibilities!

File: lib/include/client.h (hunk 1 above)
Comment (nsg) [inline lib/include/client.h]: Not sure what I was thinking, ... ignore this :)

**Comment by nsg** [comment]
@petterarvidsson you *send* direction and rotation on a player click. Why not the face that was selected and let the server figure out the meaning of it? Feels more like in style of our client server design ... or is there some advantage with this approach?

**Comment by petterarvidsson** [comment]
@nsg Basically the client has all information required to determine the direction and rotation. The direction is equal to the face selected and the rotation is the direction from which the player is pointing at the face. The server could calculate the orientation as well, but that would require a lot of code duplication between the server and the client (the client gets this information from the ray tracing that finds the selected block). I think rotation is a valid derivation of user input for the client to do, the server can still override it and do something completely different. I do believe the direction (face) and rotation (direction from which the face is pointed at) are really the two relevant parameters for block interaction in a block world and I'd say it makes sense to let the client pass them to the server, rather than the server calculating them itself. Additional thoughts?

**Comment by nsg** [comment]
I see,

If I place a block, is it considered rotated against or away from the selected face? Also to be sure, when we are talking about rotation I assume it's roll (if we talk in flight dynamic terms)?

**Comment by petterarvidsson** [comment]
@nsg Direction is which face is pointing upwards and rotation is a "roll" around the axis which now has the up and down faces. Best way to understand is to grab a test block and place it around.

**Comment by petterarvidsson** [comment]
Basically the bottom face is placed on the face that was clicked on.

**Comment by nsg** [comment]
okay, sounds good! :+1:

**Pull request merged**